awx/main/management/commands/inventory_import.py

```diff
@@ -269,12 +270,13 @@ def _build_db_instance_id_map(self):
         self.db_instance_id_map = {}
         if self.instance_id_var:
             host_qs = self.inventory_source.hosts.all()
-            host_qs = host_qs.filter(instance_id='', variables__contains=self.instance_id_var.split('.')[0])
-            for host in host_qs:
-                instance_id = self._get_instance_id(host.variables_dict)
-                if not instance_id:
-                    continue
-                self.db_instance_id_map[instance_id] = host.pk
+            for instance_id_part in reversed(self.instance_id_var.split(',')):
```

Comment:
I'm still reading through everything here, so don't have the full context yet, but had two questions come up here (mostly just for my own edification):

* Based on `self.instance_id_var.split(',')` it looks like we just weren't respecting the fact that `self.instance_id_var` could be a _list_ of potential id vars?
* Why reversed here? Does the last id var have highest precedence?

Reply:
First point - exactly, yes.

Second point - the _first_ id has the highest precedence. The loop isn't checking if the dict has the key before assigning it. So if both id vars match, it'll save the 2nd value first, but then get overwritten by the 1st.

Reply:
ah, makes sense. 👍